packages/zoe/src/types.js

```diff
@@ -28,103 +25,103 @@
- *            => Promise<OfferResultRecord<OC>>} offer
- * To redeem an invite, the user normally provides a proposal (their rules for the
- * offer) as well as payments to be escrowed by Zoe.  If either the proposal or payments
- * would be empty, indicate this by omitting that argument or passing undefined, rather
- * than passing an empty record.
- *
- * The proposal has three parts: `want` and `give` are used
- * by Zoe to enforce offer safety, and `exit` is used to specify
- * the particular payout-liveness policy that Zoe can guarantee.
- * `want` and `give` are objects with keywords as keys and amounts
- * as values. `paymentKeywordRecord` is a record with keywords as keys,
- * and the values are the actual payments to be escrowed. A payment
- * is expected for every rule under `give`.
- *
- * @property {(offerHandle: OfferHandle) => boolean} isOfferActive
- * @property {(offerHandles: OfferHandle[]) => OfferRecord[]} getOffers
- * @property {(offerHandle: OfferHandle) => OfferRecord} getOffer
- * @property {(offerHandle: OfferHandle) => Notifier<Allocation|undefined>} getOfferNotifier
- * Get a notifier (see `@agoric/notify`) for the offer's reallocations.
- * @property {(offerHandle: OfferHandle, brandKeywordRecord?: BrandKeywordRecords) => Allocation} getCurrentAllocation
- * @property {(offerHandles: OfferHandle[], brandKeywordRecord[]?: BrandKeywordRecords) => Allocation[]} getCurrentAllocations
- * @property {(installationHandle: InstallationHandle) => SourceBundle} getInstallation
- * Get the source code for the installed contract. Throws an error if the
- * installationHandle is not found.
- *
- * @typedef {Object} CompleteObj
- * @property {() => void} complete attempt to exit the contract and return a refund
+ * registering it with Zoe. Returns an installation.
  */
 
 /**
- * @template OC - the offer outcome
- * @typedef {Object} OfferResultRecord This is returned by a call to `offer` on Zoe.
- * @property {Promise<OfferHandle>} offerHandle
- * @property {Promise<PaymentPKeywordRecord>} payout A promise that resolves
- * to a record which has keywords as keys and promises for payments
- * as values. Note that while the payout promise resolves when an offer
- * is completed, the promise for each payment resolves after the remote
- * issuer successfully withdraws the payment.
- *
- * @property {Promise<OC>} outcome Note that if the offerHook throws,
- * this outcome Promise will reject, but the rest of the OfferResultRecord is
- * still meaningful.
- *
- * @property {CompleteObj} [completeObj]
- * completeObj will only be present if exitKind was 'onDemand'
+ * @typedef {Object} UserSeat
+ * @property {() => Promise<Allocation>} getAllocation
+ * @property {() => Promise<ProposalRecord>} getProposal
+ * @property {() => Promise<PaymentPKeywordRecord>} getPayouts
+ * @property {(keyword: Keyword) => Promise<Payment>} getPayout
+ * @property {() => Promise<OfferResult>} getOfferResult
+ * @property {() => void} exit
```

Comment:
I just now realized that the `exit` method of UserSeat and the `exit` method of ZCFSeat have two completely different meanings, despite looking exactly the same on two types both named `<Adjective>Seat`. All the other methods that they share have similar meaning, differing only in sync vs async, which is intuitive and shows up in the type.

It is striking because I've been through these declarations several times and I didn't notice the difference in meaning till now. Can we find distinct names for these different methods?

Reply:
Can you elaborate on the two different meanings? They both exit the seat such that the payout promise resolves, but from two different sides.

Reply:
The one of ZCFSeat is unconditional --- if its `exit` method is invoked, that seat exits.

The one on UserSeat is contingent on the exit conditions.

Reply:
I think that is not right. The `exit()` method only appears on the userSeat if the exit is unconditional.

Reply:
Nice!

In that case, could you declare the `exit` property on UserSeat to be optional?

Reply:
Yup!